src/utilities/hypre_merge_sort.c

```diff
@@ -762,7 +762,7 @@ void hypre_big_sort_and_create_inverse_map(HYPRE_BigInt *in, HYPRE_Int len, HYPR
    hypre_big_merge_sort(in, temp, len, out);
    hypre_UnorderedBigIntMapCreate(inverse_map, 2*len, 16*hypre_NumThreads());
    HYPRE_Int i;
-#ifdef HYPRE_USING_OPENMP
```

Comment:
Why do we want to restrict `omp parallelism` of this function with only hopscotch? It may not matter with the current code since this function is only called with `HYPRE_CONCURRENT_HOPSCOTCH`.

Reply:
I thought about keeping omp parallelism on this loop, however `hypre_UnorderedBigIntMapPutIfAbsent` is called inside this loop and it turns out that we need concurrent hopscotch turned on to have this loop omp parallelized in order to prevent the possibility of two threads trying to add the same key at the same time.

Reply:
Moreover, this is a separate topic but I think we could make `HYPRE_CONCURRENT_HOPSCOTCH=1`  by default when using openmp in Hypre. Then the option `--enable-hopscotch` would become obsolete. Also, all the functions provided by hopscotch are ready to work when openmp is turned off, so hopscotch is already enabled by default in hypre.

Reply:
Thanks for the clarification. I also looked at `hypre_UnorderedBigIntMapPutIfAbsent`. My impression was `CONCURRENT_HOPSCOTCH` is just an option of the implementation which doesn't affect outer loops that call it. If the calling loop can be parallelized only with concurrent hopscotch inside, we need to add some comments about it

Reply:
Yes, that's the case for enabling threaded parallelism on the calling loop. I agree that a comment about this should be added. Maybe at `hypre_hopscotch_hash.h` (where most of the hopscotch functions that can use threaded parallelism are defined)?